### DISCUSSION TITLE
Can't find previous beta release [not answered]

### DISCUSSION THREAD

**Discussion by rconnamacher** [semantic-release/semantic-release#1828] [Configuration] [answered]
Semantic-Release can't find a previous beta release on a branch that never had a 1.0.0 prod release so now the job is failing. (The branch was rebased, though I don't know if that's the culprit.) I've followed the steps in the Troubleshooting guide here: https://semantic-release.gitbook.io/semantic-release/support/troubleshooting#release-not-found-release-branch-after-git-push-force

But even after manually creating release tags matching what's already published and updating their git notes per the Troubleshooting guide, the CI job is still not seeing any prior release:

```
semantic-release:get-tags found tags for branch main: [ { gitTag: 'v1.0.0-beta.1', version: '1.0.0-beta.1', channels: [ 'beta' ] }, { gitTag: 'v1.0.0-beta.2', version: '1.0.0-beta.2', channels: [ 'beta' ] }, { gitTag: 'v1.0.0-beta.3', version: '1.0.0-beta.3', channels: [ 'beta' ] }, { gitTag: 'v1.0.0-beta.4', version: '1.0.0-beta.4', channels: [ 'beta' ] }, { gitTag: 'v1.0.0-beta.5', version: '1.0.0-beta.5', channels: [ 'beta' ] }, { gitTag: 'v1.0.0-beta.6', version: '1.0.0-beta.6', channels: [ 'beta' ] } ]

...

[semantic-release] › ℹ  No git tag version found on branch main
[semantic-release] › ℹ  No previous release found, retrieving all commits
...

[semantic-release] › ℹ  There is no previous release, the next release version is 1.0.0-beta.1

...

[semantic-release] › ✖  An error occurred while running semantic-release: Error: Command failed with exit code 128: git tag v1.0.0-beta.1 784226cf980484af6128a0e4db3ecb236a02a74a
fatal: tag 'v1.0.0-beta.1' already exists

```

The branch config is:

```
branches: [ 'release', { name: 'main', prerelease: 'beta' } ],
```

The semantic-release version is `17.4.0`.

Any idea what I missed? Or is there an issue with Semantic Release doing pre-releases before there's an initial production (v1.0.0) release?

Thanks!

**Comment by gr2m** [comment, marked as answer]
> The branch was rebased, though I don't know if that's the culprit

Yes, because semantic-release is using git notes to keep state. When you rebase, the git notes get lost and semantic-release does not know about the previous v1.0.0-beta.1 release.

**Reply by rconnamacher** [reply]
Do you know how to restore its state? Creating tags and doing the `git notes` command from the troubleshooting doc didn't seem to be enough.

**Reply by gr2m** [reply]
https://semantic-release.gitbook.io/semantic-release/support/troubleshooting#release-not-found-release-branch-after-git-push-force

**Reply by rconnamacher** [reply]
Thanks. As mentioned I had already followed those steps, and you can see the results in the above log. The debug log I included shows the git tags are there, including `channels: [ 'beta' ]` which comes from running the`git notes` step on that guide.
